src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Integer.cs

```diff
@@ -31,11 +31,12 @@ public void WriteInt64(long value)
             AdvanceDataItemCounters();
         }
 
-        public void WriteTag(CborTag tag)
+        // Writes a CBOR negative integer encoding according to
+        // https://tools.ietf.org/html/rfc7049#section-2.1
+        public void WriteCborNegativeIntegerEncoding(ulong value)
```

Comment:
I'm not sure what this even means.  `WriteCborNegativeIntegerEncoding(1)` => 0b001_0001 => `-2`?

Perhaps my question is "does this need to be public? would callers figure out how to use it correctly?"

Reply:
(And if it's public, it needs tests :smile:)

Reply:
> I'm not sure what this even means. WriteCborNegativeIntegerEncoding(1) => 0b001_0001 => -2?

Yes. this is replicating the already existing `ReadCborNegativeIntegerEncoding()`. I wasn't convinced initially but it seems that this might be necessary for encoding certain semantic values (such as BigInteger).

Reply:
Added tests